yarn/pom.xml

```diff
@@ -41,28 +42,34 @@
     <dependency>
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-yarn-api</artifactId>
+      <version>${hadoop.version}</version>
```

Comment:
right, `yarn.version`.

I agree that this feels like a worse route than #3917, as I've mentioned in a few places.

OOC, would the `yarn.version` (or `hadoop.version` -> `yarn.version`) that a given profile chooses override a `yarn.version` that was set here? I am guessing so, but am not sure.